File: app/src/main/java/com/kazakago/cleanarchitecture/domain/usecase/CityUseCaseImpl.java
package com.kazakago.cleanarchitecture.domain.usecase;

import com.kazakago.cleanarchitecture.domain.model.city.CityModel;
import com.kazakago.cleanarchitecture.domain.repository.CityRepository;

import java.io.IOException;

import io.reactivex.Observable;

/**
 * City UseCase Implement
 *
 * @author Kensuke
 */
public class CityUseCaseImpl implements CityUseCase {

    private CityRepository cityRepository;

    public CityUseCaseImpl(CityRepository cityRepository) {
        this.cityRepository = cityRepository;
    }

    public Observable<CityModel> findAll() throws IOException {
        return cityRepository.findAll();
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/domain/usecase/CityUseCaseImpl.kt
package com.kazakago.cleanarchitecture.domain.usecase

import com.kazakago.cleanarchitecture.domain.model.city.CityModel
import com.kazakago.cleanarchitecture.domain.repository.CityRepository
import io.reactivex.Observable
import java.io.IOException

/**
 * City UseCase Implement
 *
 * @author Kensuke
 */
class CityUseCaseImpl(private val cityRepository: CityRepository) : CityUseCase {

    @Throws(IOException::class)
    override fun findAll(): Observable<CityModel> {
        return cityRepository.findAll()
    }

}
